wire-schema/src/test/java/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt

```diff
@@ -0,0 +1,2341 @@
+                    "Syntax error in file.proto at 2:40: no syntax may follow trailing comment"
+            )
+        }
+    }
+
+    @Test
+    fun invalidTrailingComment() {
+        val proto = """
+                |enum Test {
+                |  FOO = 1; /
+                |}
+                """.trimMargin()
+        try {
+            ProtoParser.parse(location, proto)
+        } catch (e: IllegalStateException) {
+            assertThat(e).hasMessage(
+                    "Syntax error in file.proto at 2:12: expected '//' or '/*'"
+            )
+        }
+    }
+
+    @Test
+    fun enumValueLeadingAndTrailingCommentsAreCombined() {
+        val proto = """
+            |enum Test {
+            |  // Test all...
+            |  FOO = 1; // ...the things!
+            |}
+            """.trimMargin()
+        val parsed = ProtoParser.parse(location, proto)
+        val enumElement = parsed.types()[0] as EnumElement
+        val value = enumElement.constants()[0]
+        assertThat(value.documentation()).isEqualTo("Test all...\n...the things!")
+    }
+
+    @Test
+    fun trailingCommentNotCombinedWhenEmpty() {
+        // Can't use raw strings here; otherwise, the formatter removes the trailing whitespace on line 3.
+        val proto = "enum Test {\n" +
+                "  // Test all...\n" +
+                "  FOO = 1; //       \n" +
+                "}"
+        val parsed = ProtoParser.parse(location, proto)
+        val enumElement = parsed.types()[0] as EnumElement
+        val value = enumElement.constants()[0]
+        assertThat(value.documentation()).isEqualTo("Test all...")
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun syntaxNotRequired() {
+        val proto = "message Foo {}"
+        val parsed = ProtoParser.parse(location, proto)
+        assertThat(parsed.syntax()).isNull()
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun syntaxSpecified() {
+        val proto = """
+            |syntax = "proto3";
+            |message Foo {}
+            """.trimMargin()
+        val expected = ProtoFileElement.builder(location)
+                .syntax(ProtoFile.Syntax.PROTO_3)
+                .types(
+                        ImmutableList.of(
+                                MessageElement.builder(location.at(2, 1))
+                                        .name("Foo")
+                                        .build()
+                        )
+                )
+                .build()
+        assertThat(ProtoParser.parse(location, proto)).isEqualTo(expected)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun invalidSyntaxValueThrows() {
+        val proto = """
+                |syntax = "proto4";
+                |message Foo {}
+                """.trimMargin()
+        try {
+            ProtoParser.parse(location, proto)
+        } catch (e: IllegalStateException) {
+            assertThat(e).hasMessage("Syntax error in file.proto at 1:1: unexpected syntax: proto4")
+        }
+
```

Comment:
nit: empty lines everywhere in this file

Reply:
fixed